pkg/resourcecollector/secret.go

```diff
@@ -46,6 +46,10 @@ func (r *ResourceCollector) prepareSecretForApply(
 			secret.Annotations[serviceAccountUIDKey] = ""
 		}
 	}
+	// Reset the secret token data to empty, so that new service account token will be updated by k8s, during restore.
```

Comment:
lets empty token only if serviceaccountUID key is set ?

Reply:
Fixed